src/App.tsx

```diff
@@ -1,41 +1,141 @@
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import 'bulma/bulma.sass';
+import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-import './styles/general.scss';
+
+import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { UserSelector } from './components/UserSelector';
+import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getUsers } from './api/users';
+import { getPosts } from './api/posts';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
+export const App: React.FC = () => {
+  const [usersFromServer, setUsersFromServer] = useState<User[] | null>([]);
```

Comment:
The `usersFromServer` should always be an array. You can remove null from types
```suggestion
  const [usersFromServer, setUsersFromServer] = useState<User[]>([]);
```